Sources/SourceControl/RepositoryManager.swift

```diff
@@ -54,6 +85,8 @@ public class RepositoryManager {
         /// The status of the repository.
         fileprivate var status: Status = .uninitialized
 
+        fileprivate let serialQueue = DispatchQueue(label: "org.swift.swiftpm.repohandle-serial")
```

Comment:
Comment plz!!! This is critical stuff, this is a good place to document what the strategy is, what it is protecting, what to watch out for, etc.